solr/core/src/java/org/apache/solr/handler/GraphHandler.java

```diff
@@ -92,24 +104,29 @@ public void inform(SolrCore core) {
     }
 
     // This pulls all the overrides and additions from the config
+    List<PluginInfo> pluginInfos = core.getSolrConfig().getPluginInfos(Expressible.class.getName());
+
+    // Check deprecated approach.
     Object functionMappingsObj = initArgs.get("streamFunctions");
     if(null != functionMappingsObj){
+      log.warn("solrconfig.xml: <streamFunctions> is deprecated for adding additional streaming functions to GraphHandler.");
       NamedList<?> functionMappings = (NamedList<?>)functionMappingsObj;
       for(Entry<String,?> functionMapping : functionMappings) {
         String key = functionMapping.getKey();
         PluginInfo pluginInfo = new PluginInfo(key, Collections.singletonMap("class", functionMapping.getValue()));
-
-        if (pluginInfo.pkgName == null) {
-          Class<? extends Expressible> clazz = core.getResourceLoader().findClass((String) functionMapping.getValue(),
-              Expressible.class);
-          streamFactory.withFunctionName(key, clazz);
-        } else {
-          StreamHandler.ExpressibleHolder holder = new StreamHandler.ExpressibleHolder(pluginInfo, core, SolrConfig.classVsSolrPluginInfo.get(Expressible.class));
-          streamFactory.withFunctionName(key, () -> holder.getClazz());
-        }
-
+        pluginInfos.add(pluginInfo);
       }
+    }
 
+    for (PluginInfo pluginInfo : pluginInfos) {
+      if (pluginInfo.pkgName != null) {
+        ExpressibleHolder holder = new ExpressibleHolder(pluginInfo, core, SolrConfig.classVsSolrPluginInfo.get(Expressible.class));
+        streamFactory.withFunctionName(pluginInfo.name,
+            () -> holder.getClazz());
+      } else {
+        Class<? extends Expressible> clazz = core.getMemClassLoader().findClass(pluginInfo.className, Expressible.class);
```

Comment:
Thanks @epugh for resolving the merge conflicts! The previous `getResourceLoader` is now replaced with `getMemClassLoader` here, just checking if that's intended or accidental perhaps, not yet looked what the difference (if any) between the two methods' behaviour is.

Reply:
I'll be honest @cpoerschke I am mostly copying what is in StreamHandler, as this includes the new plugin framework support.   StreamHandler uses the `core.getMemClassLoader`, which when I click into seems to be all about plugins ;-).   My whole goal in this PR was to get StreamHandler and GraphHandler to be much more similar in how they work.

Reply:
Since this code is duplicated between Stream & Graph, can we factor it out into a common method?

Reply:
> Since this code is duplicated between Stream & Graph, can we factor it out into a common method?

Attached SOLR-13965.01.patch to the JIRA ticket.

Reply:
https://github.com/epugh/lucene-solr/pull/1 offers to merge in the current `origin/master` into this PR and then adds usage of the factored out `StreamHandler.addExpressiblePlugins` method from SOLR-13965.01.patch file.